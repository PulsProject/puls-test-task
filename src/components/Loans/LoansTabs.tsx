import React from 'react';
import Box from '@mui/material/Box';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import TabLabel from './TabLabel';
import useStyles from './LoansTabsStyles';
import { TabType, TabDetails } from './loansUtils';

interface LoansTabsProps {
  tabs: TabDetails[];
  activeTab: TabType;
  setTabValue: (value: TabType) => void;
}

const LoansTabs = ({ tabs, activeTab, setTabValue }: LoansTabsProps) => {
  const styles = useStyles();

  const handleChange = (_: React.SyntheticEvent, newValue: TabType) => {
    console.log(newValue); // TODO: remove
    setTabValue(newValue);
  };

  if (!tabs.length) return <Box sx={{ width: '100%', minHeight: '48px' }}>There&apos;s nothing to show here</Box>;

  return (
    <Box sx={{ width: '100%' }}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <Tabs value={activeTab} onChange={handleChange} aria-label="loan requests tabs">
          {tabs.map((tab) => (
            <Tab
              className={styles.loansTab}
              key={tab.tabValue}
              label={<TabLabel {...tab} />}
              value={tab.tabValue}
              id={`simple-tab-${tab.tabValue}`}
              aria-controls={`simple-tabpanel-${tab.tabValue}`}
            />
          ))}
        </Tabs>
      </Box>
    </Box>
  );
};

export default LoansTabs;
