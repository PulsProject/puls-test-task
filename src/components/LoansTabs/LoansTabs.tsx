import React from 'react';
import { Tabs, Tab, Box } from '@mui/material/';
import LoansList from '../Loans/LoansList';
import LoansTabPanel from '../LoansTabPanel/LoansTabPanel';
import Message from '../Message/Message';
// eslint-disable-next-line import/extensions
import data from '../../data/loans.json';

function tabProps(index: number) {
  return {
    id: `tab-${index}`,
    'aria-controls': `tabpanel-${index}`,
  };
}

const LoansTabs: React.FC = () => {
  const { loanRequests = [] }: any = data;
  const approvedLoans = loanRequests.filter((loan: any) => loan.status === 'waiting approval');
  const requests = loanRequests.filter((loan: any) => loan.status === 'pending settlement' || loan.status === 'to be disbursed');
  const activeLoans = loanRequests.filter((loan: any) => loan.status === 'active');
  const closedLoans = loanRequests.filter((loan: any) => loan.status === 'closed');
  const rejectedLoans = loanRequests.filter((loan: any) => loan.status === 'rejected');
  const [value, setValue] = React.useState(0);

  const approvedTabLabel = `Approved Loans ${activeLoans.length}`;
  const requestsTabLabel = `Requests ${requests.length}`;
  const activeTabLabel = `Active Loans ${activeLoans.length}`;
  const closedTabLabel = `Closed Loans ${closedLoans.length}`;
  const rejectedTabLabel = `Rejected Loans ${rejectedLoans.length}`;

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };

  if (!loanRequests.length) {
    return <Message />;
  }

  return (
        <>
            <Box sx={{ width: '100%' }}>
                <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
                    <Tabs value={value} onChange={handleChange} aria-label="loans tabs">
                        {!!approvedLoans.length && <Tab label={approvedTabLabel} {...tabProps(0)} />}
                        {!!requests.length && <Tab label={requestsTabLabel} {...tabProps(1)} />}
                        {!!activeLoans.length && <Tab label={activeTabLabel} {...tabProps(2)} />}
                        {!!closedLoans.length && <Tab label={closedTabLabel} {...tabProps(3)} />}
                        {!!rejectedLoans.length && <Tab label={rejectedTabLabel} {...tabProps(4)} />}
                    </Tabs>
                </Box>
                <LoansTabPanel value={value} index={0}>
                    <LoansList loanRequests={approvedLoans}/>
                </LoansTabPanel>
                <LoansTabPanel value={value} index={1}>
                    <LoansList loanRequests={requests}/>
                </LoansTabPanel>
                <LoansTabPanel value={value} index={2}>
                    <LoansList loanRequests={activeLoans}/>
                </LoansTabPanel>
                <LoansTabPanel value={value} index={3}>
                    <LoansList loanRequests={closedLoans}/>
                </LoansTabPanel>
                <LoansTabPanel value={value} index={4}>
                    <LoansList loanRequests={rejectedLoans}/>
                </LoansTabPanel>
            </Box>
        </>
  );
};

export default LoansTabs;