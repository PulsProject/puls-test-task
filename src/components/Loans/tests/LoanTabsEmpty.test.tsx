import renderer from 'react-test-renderer';

import { LoansTabs } from '../LoansTabs';

it('renders <LoansTabs/> with no data', () => {
  const tree = renderer.create(<LoansTabs />).toJSON();
  expect(tree).toMatchSnapshot();
});
