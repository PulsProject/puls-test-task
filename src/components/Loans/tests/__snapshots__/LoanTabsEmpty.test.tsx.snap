// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <LoansTabs/> with no data 1`] = `
<div
  className="MuiBox-root css-0"
>
  There's nothing to show here
</div>
`;
